rules_test.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"testing"
```

Comment:
Like common advice: let's use https://github.com/stretchr/testify for testing. Almost every golang project uses it, this is standard de-facto.

Reply:
muuuuuuuuuuuuch better than the default testing